src/Cache.cpp

```diff
@@ -115,19 +115,19 @@ tr1::shared_ptr<Frame> Cache::GetSmallestFrame()
 	const GenericScopedLock<CriticalSection> lock(*cacheCriticalSection);
 	tr1::shared_ptr<openshot::Frame> f;
 
-	 // Loop through frame numbers
-	 deque<long int>::iterator itr;
-	 long int smallest_frame = -1;
-	 for(itr = frame_numbers.begin(); itr != frame_numbers.end(); ++itr)
-	 {
-		 if (*itr < smallest_frame || smallest_frame == -1)
-			 smallest_frame = *itr;
-	 }
+	// Loop through frame numbers
+	deque<long int>::iterator itr = frame_numbers.begin();
+	long int smallest_frame = *itr;
```

Comment:
I can't see any problem anywhere else in Cache.cpp - frame_numbers.begin() is always used in for cycle with termination condition `itr != frame_numbers.end()` that will terminate the loop on first check with empty frame_numbers, so it won't dereference in the cycle's body.